File: 2021-10/최소직사각형.js
function solution(sizes) {
    return sizes.map(arr => arr.sort((a, b) => b - a))
        .reduce((acc, cur) => [Math.max(acc[0], cur[0]), Math.max(acc[1], cur[1])])
        .reduce((acc, cur) => acc * cur);
}

File: programmers/src/main/java/level1/monthlycodechallenge/최소직사각형.java
package level1.monthlycodechallenge;

public class 최소직사각형 {

	public int solution(int[][] sizes) {
		int w = 0;
		int h = 0;

		for (int i = 0; i < sizes.length; i++) {
			int a = sizes[i][0];
			int b = sizes[i][1];
			if (a < b) {
				sizes[i][0] = b;
				sizes[i][1] = a;
			}
			w = Math.max(w, sizes[i][0]);
			h = Math.max(h, sizes[i][1]);
		}
		return w * h;
	}

}
